yarn-project/foundation/src/eth-address/index.ts

```diff
@@ -174,14 +174,22 @@ export class EthAddress {
     return EthAddress.toChecksumAddress(this.buffer.toString('hex'));
   }
 
+  /**
+   * Alias for toBuffer32.
+   * @returns A 32-byte Buffer containing the padded Ethereum address.
+   */
+  public toBuffer() {
+    return this.toBuffer32();
+  }
```

Comment:
Serializing EthAddress to 20 bytes by default was causing issues with serialization to/from cpp (yay). Msgpack autobindings were calling toBuffer on this field for serializing, which returned 20 bytes, but cpp was expecting 32. This resulted in the eth address being left or right aligned in cpp depending on how it got there. This was not uncovered before because eth address deserialization handled both cases, but when we started hashing this field in cpp, the difference started to show.

Please add 4hs to the counter of "time lost due to serialization issues"!

Reply:
💀💀💀

Reply:
💀💀💀